### PR TITLE
Rewrite ARM IOMD/VIDC20 chips into own devices, merged implementations from ssfindo.cpp & riscpc.cpp [Angelo Salese]

### PR DESCRIPTION
**Pull request by angelosa**
* riscpc now enters in desktop mode with a valid nvram;
* adds preliminary sound for ppcar;

### CODE REVIEW

**Comment by angelosa** [comment]
**Not yet ready for inclusion**
I don't understand how i2c bothers about the order that outputs are written to, it sounds like something that is flawed in the core itself.

File: src/devices/machine/acorn_vidc.cpp [hunk 1]
```diff
@@ -150,6 +159,7 @@ void acorn_vidc10_device::device_start()
 	save_item(NAME(m_sound_frequency_test_bit));
 	save_item(NAME(m_cursor_enable));
 	save_pointer(NAME(m_crtc_regs), CRTC_VCER+1);
+	save_pointer(NAME(m_crtc_raw_horz), 2);
 	m_data_vram = auto_alloc_array_clear(machine(), u8, m_data_vram_size);
```
Comment (ajrhacker) [inline src/devices/machine/acorn_vidc.cpp]: The `auto_alloc` functions are considered deprecated; `make_unique_clear` should be used here instead.

File: src/devices/machine/acorn_vidc.h [hunk 2]
```diff
@@ -63,15 +63,39 @@ class acorn_vidc10_device : public device_t,
...
+	inline void screen_vblank_line_update();
+	inline void screen_dynamic_res_change();
+	void refresh_sound_frequency();
+
+	u16 m_pal_4bpp_base;
+	u16 m_pal_cursor_base;
+	u16 m_pal_border_base;
+	const double m_sound_internal_divider = 8.0;
+
+	uint8_t m_bpp_mode, m_crtc_interlace;
```
Comment (ajrhacker) [inline src/devices/machine/acorn_vidc.h]: Mixing the `uX` shorthand types with `uintX_t` is highly inconsistent.

File: src/devices/machine/acorn_vidc.cpp (hunk 1 above)
Comment (angelosa) [reply src/devices/machine/acorn_vidc.cpp]: Not touched by the PR, additionally I have no idea about how to pass a std::unique_ptr to a function in C++ (cfr. `draw()` ).

Comment (MooglyGuy) [reply src/devices/machine/acorn_vidc.cpp]: Pass it either as a reference or as a bare pointer, passing in &my_unique_ptr[0] from the calling code.

**Comment by cuavas** [comment]
Nice cleanup getting all of this out of drivers and de-duplicating it.

**Pull request merged**